### ISSUE TITLE
Post process of parsing plist fails when disk configured as FusionDrive connected.

### ISSUE THREAD

**Issue by ailen0ada** [resin-io-modules/drivelist#258]
When FusionDrive exists, `diskutil list -plist` outputs following dictionary (a part) :

```xml
		<dict>
			<key>Content</key>
			<string>Apple_HFS</string>
			<key>DeviceIdentifier</key>
			<string>disk2</string>
			<key>MountPoint</key>
			<string>/</string>
			<key>Size</key>
			<integer>2114568323072</integer>
			<key>VolumeName</key>
			<string>Macintosh HD</string>
		</dict>
```

As a consequence, `layout.Partitions.concat` will fail due to `Partitions` is `undefined`.
https://github.com/resin-io-modules/drivelist/blob/5beaa42dda76e7a2f3c11414b34840b4fd9370fb/lib/diskutil.js#L122

**Comment by ailen0ada** [comment]
After some investigation, it seems the immediate cause is `CoreStorage` configuration. Problem occurs on MacBookPro Late2016 with 1TB SSD.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>AllDisks</key>
	<array>
		<string>disk0</string>
		<string>disk0s1</string>
		<string>disk0s2</string>
		<string>disk0s3</string>
		<string>disk1</string>
		<string>disk2</string>
		<string>disk2s1</string>
	</array>
	<key>AllDisksAndPartitions</key>
	<array>
		<dict>
			<key>Content</key>
			<string>GUID_partition_scheme</string>
			<key>DeviceIdentifier</key>
			<string>disk0</string>
			<key>Partitions</key>
			<array>
				<dict>
					<key>Content</key>
					<string>EFI</string>
					<key>DeviceIdentifier</key>
					<string>disk0s1</string>
					<key>DiskUUID</key>
					<string>B5359E11-98F8-4256-A70B-F139D6272E2E</string>
					<key>Size</key>
					<integer>314572800</integer>
					<key>VolumeName</key>
					<string>EFI</string>
					<key>VolumeUUID</key>
					<string>E783267B-A4C3-3556-B751-DBED770EB996</string>
				</dict>
				<dict>
					<key>Content</key>
					<string>Apple_CoreStorage</string>
					<key>DeviceIdentifier</key>
					<string>disk0s2</string>
					<key>DiskUUID</key>
					<string>E421C13C-F1DA-466C-80E2-5EBC5580F280</string>
					<key>Size</key>
					<integer>999590961152</integer>
				</dict>
				<dict>
					<key>Content</key>
					<string>Apple_Boot</string>
					<key>DeviceIdentifier</key>
					<string>disk0s3</string>
					<key>DiskUUID</key>
					<string>9D228416-2FD1-4D62-B94F-95AF93F64351</string>
					<key>Size</key>
					<integer>650002432</integer>
					<key>VolumeName</key>
					<string>Recovery HD</string>
					<key>VolumeUUID</key>
					<string>52DAB313-9F5E-32EB-8C02-206E19FDC533</string>
				</dict>
			</array>
			<key>Size</key>
			<integer>1000555581440</integer>
		</dict>
		<dict>
			<key>Content</key>
			<string>Apple_HFS</string>
			<key>DeviceIdentifier</key>
			<string>disk1</string>
			<key>MountPoint</key>
			<string>/</string>
			<key>Size</key>
			<integer>999250984960</integer>
			<key>VolumeName</key>
			<string>Macintosh HD</string>
		</dict>
		<dict>
			<key>Content</key>
			<string>FDisk_partition_scheme</string>
			<key>DeviceIdentifier</key>
			<string>disk2</string>
			<key>Partitions</key>
			<array>
				<dict>
					<key>Content</key>
					<string>Windows_NTFS</string>
					<key>DeviceIdentifier</key>
					<string>disk2s1</string>
					<key>MountPoint</key>
					<string>/Volumes/HDD03</string>
					<key>Size</key>
					<integer>2000398932992</integer>
					<key>VolumeName</key>
					<string>HDD03</string>
					<key>VolumeUUID</key>
					<string>10D703DF-7323-3D58-B7B9-9A18FEFB27EB</string>
				</dict>
			</array>
			<key>Size</key>
			<integer>2000398934016</integer>
		</dict>
	</array>
	<key>VolumesFromDisks</key>
	<array>
		<string>Macintosh HD</string>
		<string>HDD03</string>
	</array>
	<key>WholeDisks</key>
	<array>
		<string>disk0</string>
		<string>disk1</string>
		<string>disk2</string>
	</array>
</dict>
</plist>
```

**Comment by jhermsmeier** [comment]
Neat, thanks for the additional info! Working on a fix :)

**Comment by jhermsmeier** [comment]
@ailen0ada can I ask you what your Mac OS version is, and what `diskutil info -plist /dev/disk1` outputs for you?

**Comment by ailen0ada** [comment]
running on 10.12.6, and following is output of `diskutil info -plist /devdisk1`.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Bootable</key>
	<true/>
	<key>BusProtocol</key>
	<string>PCI-Express</string>
	<key>CanBeMadeBootable</key>
	<false/>
	<key>CanBeMadeBootableRequiresDestroy</key>
	<false/>
	<key>Content</key>
	<string>Apple_HFS</string>
	<key>CoreStorageCompositeDisk</key>
	<false/>
	<key>CoreStorageEncrypted</key>
	<true/>
	<key>CoreStorageLVFUUID</key>
	<string>53079938-48AC-4B2F-AD3C-A3CDB716710C</string>
	<key>CoreStorageLVGUUID</key>
	<string>A2BB510B-EF45-42A4-BC03-2B511B521841</string>
	<key>CoreStorageLVUUID</key>
	<string>02787797-CC55-48C6-BB51-BE7839DB7D40</string>
	<key>CoreStoragePVs</key>
	<array>
		<dict>
			<key>CoreStoragePVDisk</key>
			<string>disk0s2</string>
			<key>CoreStoragePVUUID</key>
			<string>74E12A48-A5F7-4849-8169-D2F4C65D153E</string>
		</dict>
	</array>
	<key>DeviceBlockSize</key>
	<integer>4096</integer>
	<key>DeviceIdentifier</key>
	<string>disk1</string>
	<key>DeviceNode</key>
	<string>/dev/disk1</string>
	<key>DeviceTreePath</key>
	<string>IODeviceTree:/PCI0@0/RP17@1B/SSD0@0/IONVMeController</string>
	<key>DiskUUID</key>
	<string>02787797-CC55-48C6-BB51-BE7839DB7D40</string>
	<key>Ejectable</key>
	<false/>
	<key>EjectableMediaAutomaticUnderSoftwareControl</key>
	<false/>
	<key>EjectableOnly</key>
	<false/>
	<key>FilesystemName</key>
	<string>Journaled HFS+</string>
	<key>FilesystemType</key>
	<string>hfs</string>
	<key>FilesystemUserVisibleName</key>
	<string>Mac OS Extended (Journaled)</string>
	<key>FreeSpace</key>
	<integer>519507255296</integer>
	<key>GlobalPermissionsEnabled</key>
	<true/>
	<key>IOKitSize</key>
	<integer>999250984960</integer>
	<key>IORegistryEntryName</key>
	<string>Macintosh HD</string>
	<key>Internal</key>
	<true/>
	<key>JournalOffset</key>
	<integer>455176192</integer>
	<key>JournalSize</key>
	<integer>83886080</integer>
	<key>LowLevelFormatSupported</key>
	<false/>
	<key>MediaName</key>
	<string>APPLE SSD SM1024L</string>
	<key>MediaType</key>
	<string>Generic</string>
	<key>MountPoint</key>
	<string>/</string>
	<key>OS9DriversInstalled</key>
	<false/>
	<key>ParentWholeDisk</key>
	<string>disk1</string>
	<key>RAIDMaster</key>
	<false/>
	<key>RAIDSlice</key>
	<false/>
	<key>RecoveryDeviceIdentifier</key>
	<string>disk0s3</string>
	<key>Removable</key>
	<false/>
	<key>RemovableMedia</key>
	<false/>
	<key>RemovableMediaOrExternalDevice</key>
	<false/>
	<key>SMARTStatus</key>
	<string>Not Supported</string>
	<key>Size</key>
	<integer>999250984960</integer>
	<key>SolidState</key>
	<true/>
	<key>SupportsGlobalPermissionsDisable</key>
	<true/>
	<key>SystemImage</key>
	<false/>
	<key>TotalSize</key>
	<integer>999250984960</integer>
	<key>VirtualOrPhysical</key>
	<string>Virtual</string>
	<key>VolumeAllocationBlockSize</key>
	<integer>4096</integer>
	<key>VolumeName</key>
	<string>Macintosh HD</string>
	<key>VolumeSize</key>
	<integer>999250984960</integer>
	<key>VolumeUUID</key>
	<string>06421672-F73B-3A55-87D4-B74BEDA6C1C5</string>
	<key>WholeDisk</key>
	<true/>
	<key>Writable</key>
	<true/>
	<key>WritableMedia</key>
	<true/>
	<key>WritableVolume</key>
	<true/>
</dict>
</plist>
```

**Comment by jhermsmeier** [comment]
@ailen0ada @johanpoirier fix released as drivelist@6.1.2

**Comment by johanpoirier** [comment]
>  fix released as drivelist@6.1.2

Awesome, thanks!
